Sources/WalletConnectModal/Modal/ModalViewModel.swift

```diff
@@ -164,79 +170,104 @@ final class ModalViewModel: ObservableObject {
             // Small deliberate delay to ensure animations execute properly
             try await Task.sleep(nanoseconds: 500_000_000)
                 
-            withAnimation {
-                self.wallets = wallets.sorted {
-                    guard let lhs = $0.order else {
-                        return false
-                    }
-                        
-                    guard let rhs = $1.order else {
-                        return true
-                    }
-                    
-                    return lhs < rhs
-                }
-                
-                loadRecentWallets()
-            }
+            self.wallets = wallets
+            
+            checkInstalledWallets()
+            loadRecentWallets()
         } catch {
             toast = Toast(style: .error, message: error.localizedDescription)
         }
     }
 }
 
-// MARK: - Recent Wallets
+// MARK: - Sorting and filtering
 
-private extension ModalViewModel {
+private extension Array where Element: Listing {
     
-    func sortByRecent(_ input: [Listing]) -> [Listing] {
-        input.sorted { lhs, rhs in
-            guard let lhsLastTimeUsed = lhs.lastTimeUsed else {
+    func sortByOrder() -> [Listing] {
+        self.sorted {
```

Comment:
Actually, since it is in a single computed property like this, which does not sort in place but returns a new ordered sequence, it should trigger a single UI update once it orders everything.

```swift
var filteredWallets: [Listing] {
        wallets
            .sortByOrder()
            .sortByInstalled()
            .sortByRecent()
            .filter(searchTerm: searchTerm)
}
 ```
 
 One concern might be that we are doing unnecessary iterations since `order` and `installed` won't change, so I will extract those to run only once.